src/simpleseqplay.cc

```diff
@@ -146,22 +150,48 @@ namespace dynamicgraph
           ("SimpleSeqPlay posture: Signals not initialized. read files first.");
         }
         std::size_t configId;
+	// If we are still waiting to start
         if (state_ == 0)
-        {
-          configId = 0;
-        }
-        else if (state_ == 1)
-        {
-          configId = t - startTime_;
-          if (configId == posture_.size () - 1)
-          {
-            state_ = 2;
-          }
-        }
+	  {
+	    // return the current posture.
+	    pos = currentPostureSIN_.access(t);
+	    return pos;
+	  }
+	// Going to the first position
+	else if (state_ == 1)
+	  {
+	    // Compute the difference between current posture and desired one.
+	    dg::Vector deltapos = posture_[0]-currentPostureSIN_.access(t);
+
+	    // If sufficiently closed to the first posture of the seqplay.
+	    if (deltapos.norm()<1e-4)
+	      {
+		// Switch to the next state.
+		state_=2;
+		startTime_ = postureSOUT_.getTime();
+		pos = posture_ [0];
+	      }
+	    else
+	      {
+		// Tries to go closer to the first posture.
+		deltapos = (deltapos * dt_)/time_to_start_; 
```

Comment:
it should be `deltapos = (deltapos * )/time_to_start_;`, since we are already simulating the entity after every dt seconds